README.md

```diff
@@ -6,19 +6,17 @@ Most messages sent from the base chain to Fuel will use a predicate as the messa
 
 The Message to Contract Predicate is for messages that are trying to send a data payload to a designated Fuel contract. This predicate runs through the following checks:
 - Verify Inputs
-  - There is exactly one `InputCoin` and it has `asset_id` set as the base asset to cover gas costs
-  - There is exactly one `InputMessage` (the message with a data payload to send to a contract)
-  - There is at least one `InputContract` with `contractID` that matches the first 32 bytes in the message data field
-  - Additional `InputContract` are allowed but are not required
+  - There are only 3 total inputs
+  - The first input is an `InputContract` with `contractID` that matches the first 32 bytes in the message data field
+  - The second input is an `InputMessage` and is the only `InputMessage` (the message with a data payload to send to a contract)
+  - The third input is an `InputCoin` with `asset_id` set as the base asset (to cover gas costs)
```

Comment:
Is it actually needed to check the asset ID is the base asset? By construction of the protocol, this is guaranteed.

Reply:
Why is it guaranteed? The input message value could be used to pay for gas and this coin input could be something else. We could instead just check that the change output is of base asset id, but the GTF opcode currently does not support getting a change output asset id